Add HAProxy Ingress Controller v1.4 chart

dcorbett-haproxy opened the pull request:
Add chart for HAProxy Ingress Controller v1.4

PennyScissors commented:
I added this chart as a custom catalog in Rancher 2.4, but I haven't been able to deploy it. Is there an initial setup or requirements required to deploy this? Please take me through the steps you took to test it if possible, thank you!

I'm running into this error:
`Wait helm template failed. 2020/04/14 19:45:53 Warning: Merging destination map for chart 'haproxy'. Overwriting table item 'image', with non table value: true Error: render error in "haproxy/templates/controller-deployment.yaml": template: haproxy/templates/controller-deployment.yaml:51:28: executing "haproxy/templates/controller-deployment.yaml" at <.Values.controller.image.repository>: can't evaluate field repository in type interface {} : exit status 1`

dcorbett-haproxy commented:
> I added this chart as a custom catalog in Rancher 2.4, but I haven't been able to deploy it. Is there an initial setup or requirements to deploy this? Please take me through the steps you took to test it if possible, thank you!
>

My apologies on this. I had a typo within a variable in questions.yaml that was conflicting with values.yaml. I had fixed this previously internally but forgot to actually merge that change before submission. Sorry about that. Everything should be good now, can you please try again?

PennyScissors commented on charts/haproxy/v1.4/Chart.yaml:
```diff
@@ -0,0 +1,35 @@
+
+apiVersion: v1
+name: haproxy
+version: 1.1.1
```
The parent folder v1.4/ has to be renamed to v1.1.1/
Is named after the chart version indicated here rather than the version of the app it wraps.

dcorbett-haproxy replied:
> The parent folder v1.4/ has to be renamed to v1.1.1/
> Is named after the chart version indicated here rather than the version of the app it wraps.

Ah, thank you for this info. I have updated.

PennyScissors commented:
> > I added this chart as a custom catalog in Rancher 2.4, but I haven't been able to deploy it. Is there an initial setup or requirements to deploy this? Please take me through the steps you took to test it if possible, thank you!
> 
> My apologies on this. I had a typo within a variable in questions.yaml that was conflicting with values.yaml. I had fixed this previously internally but forgot to actually merge that change before submission. Sorry about that. Everything should be good now, can you please try again?

No worries. I was able to successfully deploy it this time, so it should be good to go after the change I requested.

guangbochen commented:
@dcorbett-haproxy can u please help to squash the commits into 2, thanks.

dcorbett-haproxy commented:
> @dcorbett-haproxy can u please help to squash the commits into 2, thanks.

Done.
